store_test.go

```diff
@@ -306,6 +311,24 @@ func testSessionFailure(t *testing.T, ctx context.Context, store *Store) {
 	t.Log(err)
 }
 
+func testWalkFromUnsetKey(t *testing.T, ctx context.Context, store *Store) {
+	err := store.Walk(ctx, "foo/bar/kada/bra", func(key string) error {
+		return errors.New("should to walk through " + key)
```

Comment:
Try to rephrase your statements here to be assertive instead of saying "should", there's a lot of value in being very explicit about what we expect and _should_ doesn't communicate this well.

For example here you could say "the Walk callback was called for a key that did not exist". It tells the developer exactly what went wrong instead of leaving details up to interpretation of what _should_ have happened.